src/services/catalog.service.ts

```diff
@@ -145,26 +145,56 @@ export class ImageItem implements IServiceItem {
   public description: string;
   public longDescription: string;
   public resource: any;
-  public isBuilder: boolean;
-  public isHidden: boolean;
+  private validSpecTagName: any;
   private tags: any;
   private catalogSrv: CatalogService;
 
   constructor (image: any, catalogSrv : CatalogService) {
     this.resource = image;
     this.catalogSrv = catalogSrv;
-    this.tags = this.catalogSrv.$filter('imageStreamTagTags')(this.resource);
-    this.iconClass = this.getIcon();
-    this.name = this.getName();
-    this.description = this.getDescription();
-    this.longDescription = this.getLongDescription();
-    this.catsBySubCats = this.getCategoriesBySubCategories();
-    this.isBuilder = _.includes(this.tags, 'builder');
-    this.isHidden = _.includes(this.tags, 'hidden');
+    this.validSpecTagName = this.getValidSpecTagName();
+    if (this.validSpecTagName) {
+      this.tags = this.getTags();
+      this.iconClass = this.getIcon();
+      this.name = this.getName();
+      this.description = this.getDescription();
+      this.longDescription = this.getLongDescription();
+      this.catsBySubCats = this.getCategoriesBySubCategories();
```

Comment:
I'm making a general statement that we should try to use API objects directly when possible. This is the approach we took in origin-web-console, and it worked well.

ServiceItem and ImageItem makes sense here since we're showing different types as catalog tiles with a generic interface. But I think when we pass the objects to the wizard, we should pass the API object instead. We'll probably need different components anyway for the two flows (builder and service class) since they're so different, so the abstraction is less helpful there.